Source/items.cpp

```diff
@@ -487,26 +487,36 @@ void CalcSelfItems(Player &player)
 	do {
 		changeflag = false;
 		auto *pi = player.InvBody;
+		int tsa = 0;
+		int tma = 0;
+		int tda = 0;
 		for (int i = 0; i < NUM_INVLOC; i++, pi++) {
 			if (!pi->isEmpty() && pi->_iStatFlag) {
 				bool sf = true;
-				if (sa + player._pBaseStr < pi->_iMinStr)
+				// cap stats to 0
+				int currstr = std::max(0, sa + player._pBaseStr);
+				int currmag = std::max(0, ma + player._pBaseMag);
+				int currdex = std::max(0, da + player._pBaseDex);
+				if (currstr < pi->_iMinStr)
 					sf = false;
-				if (ma + player._pBaseMag < pi->_iMinMag)
+				if (currmag < pi->_iMinMag)
 					sf = false;
-				if (da + player._pBaseDex < pi->_iMinDex)
+				if (currdex < pi->_iMinDex)
 					sf = false;
 				if (!sf) {
 					changeflag = true;
 					pi->_iStatFlag = false;
 					if (pi->_iIdentified) {
-						sa -= pi->_iPLStr;
-						ma -= pi->_iPLMag;
-						da -= pi->_iPLDex;
+						tsa -= pi->_iPLStr;
+						tma -= pi->_iPLMag;
+						tda -= pi->_iPLDex;
 					}
 				}
 			}
 		}
+		sa += tsa;
+		ma += tma;
+		da += tda;

```

Comment:
Simpler version :)

```cpp
		// cap stats to 0
		int currstr = std::max(0, sa + player._pBaseStr);
		int currmag = std::max(0, ma + player._pBaseMag);
		int currdex = std::max(0, da + player._pBaseDex);

		for (auto &equipment : player.InvBody) {
			if (equipment.isEmpty() || !equipment._iStatFlag)
				continue;

			if (currstr >= equipment._iMinStr && currmag >= equipment._iMinMag && currdex >= equipment._iMinDex)
				continue;

			changeflag = true;
			equipment._iStatFlag = false;
			if (equipment._iIdentified) {
				sa -= equipment._iPLStr;
				ma -= equipment._iPLMag;
				da -= equipment._iPLDex;
			}
		}
```

Reply:
Bonus points for also making the first loop ranged based.

Reply:
> Simpler version :)
> 
> ```c++
> 		// cap stats to 0
> 		int currstr = std::max(0, sa + player._pBaseStr);
> 		int currmag = std::max(0, ma + player._pBaseMag);
> 		int currdex = std::max(0, da + player._pBaseDex);
> 
> 		for (auto &equipment : player.InvBody) {
> 			if (equipment.isEmpty() || !equipment._iStatFlag)
> 				continue;
> 
> 			if (currstr >= equipment._iMinStr && currmag >= equipment._iMinMag && currdex >= equipment._iMinDex)
> 				continue;
> 
> 			changeflag = true;
> 			equipment._iStatFlag = false;
> 			if (equipment._iIdentified) {
> 				sa -= equipment._iPLStr;
> 				ma -= equipment._iPLMag;
> 				da -= equipment._iPLDex;
> 			}
> 		}
> ```

So simple that it doesn't even work XD